gio/src/auto/async_result.rs

```diff
@@ -0,0 +1,71 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use glib::object::IsA;
+use glib::translate::*;
+use std::fmt;
+use std::ptr;
+
+glib::wrapper! {
+    pub struct AsyncResult(Interface<ffi::GAsyncResult>);
+
+    match fn {
+        get_type => || ffi::g_async_result_get_type(),
+    }
+}
+
+pub const NONE_ASYNC_RESULT: Option<&AsyncResult> = None;
+
+pub trait AsyncResultExt: 'static {
+    #[doc(alias = "g_async_result_get_source_object")]
+    fn get_source_object(&self) -> Option<glib::Object>;
+
+    //#[doc(alias = "g_async_result_get_user_data")]
+    //fn get_user_data(&self) -> /*Unimplemented*/Option<Fundamental: Pointer>;
+
+    //#[doc(alias = "g_async_result_is_tagged")]
+    //fn is_tagged(&self, source_tag: /*Unimplemented*/Option<Fundamental: Pointer>) -> bool;
+
+    #[doc(alias = "g_async_result_legacy_propagate_error")]
+    fn legacy_propagate_error(&self) -> Result<(), glib::Error>;
+}
+
+impl<O: IsA<AsyncResult>> AsyncResultExt for O {
+    fn get_source_object(&self) -> Option<glib::Object> {
+        unsafe {
+            from_glib_full(ffi::g_async_result_get_source_object(
+                self.as_ref().to_glib_none().0,
+            ))
+        }
+    }
+
+    //fn get_user_data(&self) -> /*Unimplemented*/Option<Fundamental: Pointer> {
```

Comment:
How would one do this in bindings usually? A subclass of `GAsyncResult`?

Reply:
In Rust bindings, associating some data to the Task callback can be done with a closure with the `user_data` moved into its scope, without the need to explicitly pass a `user_data` pointer to the Task constructor. I will add this to the `gio_task.rs` example